### PR TITLE
get resource changes after date API

### PR DESCRIPTION
**Pull request by adropofilm**
**Objective**
1. Created `GET /changes?updated_after={date_time}` endpoint to get all resources updated after `date_time` + error handling when invalid datetime format given.
2. Minor changes in README to reflect ruby version update + documentation details

**Questions**
1. Any tips on how I can get `resource_id` to be passed in as an optional query param, but still keeping the grabbing of `date_time` maintainable (date_time is filtered via scope)? Tried to see how yall were filtering params, etc now but got lost (still new to rails/ruby), k
2. Where are  unit tests located? I tried to find them but couldn't, thought they'd be in /test but no such directory
3. Are the resource data being shown in `app/views/api/v4/resources/changes.json.streamer` fine? Not familiar with what people who'd use this endpoint would want to know besides what's there.

**Resolves**: https://github.com/quran/quran.com-api/issues/575

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:49180

File: app/controllers/api/v4/resources_controller.rb [hunk 1]
```diff
@@ -106,5 +106,17 @@ def languages
 
       render
     end
+
+    def changes
+      updated_after = params[:updated_after]
+      @resources = ResourceContent.changes(updated_after)
```
Comment (naveed-ahmad) [inline app/controllers/api/v4/resources_controller.rb:112]: How about following implementation? 

```
if time = after_timestamp
   @resources = ResourceContent.changes(after: time)
   render 
else 
  render_422("Invalid datetime format")
end 
```

```
def after_timestamp
  if time = params[:updated_after].presence
    time !~ /\D/ ? Time.at(time) : Time.parse(time)
  end
end
```

`updated_after` could be a date/time or unix time stamp.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -54,4 +58,11 @@ def eager_load_translated_name(records)
         .order('translated_names.language_priority DESC')
     end
   end
+
+  def valid_time?(datetime)
```
Comment (naveed-ahmad) [inline app/controllers/application_controller.rb:62]: see above comment

File: app/models/resource_content.rb [hunk 3]
```diff
@@ -57,6 +57,8 @@ class ResourceContent < ApplicationRecord
   scope :one_chapter, -> { where cardinality_type: CardinalityType::OneChapter }
   scope :approved, -> { where approved: true }
   scope :recitations, -> { where sub_type: SubType::Audio }
+  scope :changes, ->(updated_after) { where("updated_at > ?", updated_after) }
```
Comment (naveed-ahmad) [inline app/models/resource_content.rb:60]: Class method makes more sense here.

**Review by naveed-ahmad — CHANGES_REQUESTED** [review]
JAK for your first PR! Some minor issues.

File: app/views/api/v4/resources/changes.json.streamer [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+                      :name,
+                      :updated_at,
+                      :resource_type,
+                      :resource_type_name,
+                      :cardinality_type,
+                      :description,
+                      :language_name,
+                      :resource_info,
+                      :slug,
+                      :sub_type
```
[the remark this answers is not in the document]
Comment (adropofilm) [reply app/views/api/v4/resources/changes.json.streamer:16]: fixed

File: app/models/resource_content.rb (hunk 3 above)
Comment (adropofilm) [reply app/models/resource_content.rb:60]: fixed

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (adropofilm) [reply app/controllers/application_controller.rb:62]: Fixed

File: app/controllers/api/v4/resources_controller.rb (hunk 1 above)
Comment (adropofilm) [reply app/controllers/api/v4/resources_controller.rb:112]: Fixed

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:49181

**Review by naveed-ahmad — APPROVED** [review]

**Pull request merged**